src/test/fuzz/util.h

```diff
@@ -470,4 +470,63 @@ void ReadFromStream(FuzzedDataProvider& fuzzed_data_provider, Stream& stream) no
     }
 }
 
+class FuzzedSocket : public Socket
+{
+    FuzzedDataProvider& fuzzed_data_provider;
```

Comment:
nit: `m_fuzzed_data_provider` so the reference in the constructor can just be `fuzzed_data_provider`?

Reply:
Good nit! Adressed!